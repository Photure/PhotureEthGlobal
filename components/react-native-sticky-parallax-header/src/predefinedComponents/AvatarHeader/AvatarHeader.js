import React from 'react';
import {
  Text,
  View,
  Image,
  TouchableOpacity,
  Animated,
  StatusBar,
  ViewPropTypes,
  ScrollView,
} from 'react-native';
import {
  func,
  string,
  number,
  bool,
  oneOfType,
  oneOf,
  instanceOf,
  element,
  shape,
  node,
} from 'prop-types';
import StickyParallaxHeader from '../../StickyParallaxHeader';
import { constants, sizes } from '../../constants';
import styles from './AvatarHeader.styles';
import { Brandon } from '../../assets/data/cards';
import RenderContent from './defaultProps/defaultProps';

import { SvgUri } from 'react-native-svg';

const { event, ValueXY } = Animated;

const finishImgPosition = 31;
const startImgPosition = 27;

class AvatarHeader extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      headerLayout: {
        height: 0,
      },
    };
    this.scrollY = new ValueXY();
  }

  setHeaderSize = (headerLayout) => {
    const { headerSize } = this.props;
    if (headerSize) headerSize(headerLayout);
    this.setState({ headerLayout });
  };

  scrollPosition(value) {
    const {
      headerLayout: { height },
    } = this.state;

    return constants.scrollPosition(height, value);
  }

  renderAvatarHeader = () => {
    const {
      leftTopIconOnPress,
      leftTopIcon,
      rightTopIcon,
      rightTopIconOnPress,
      walletAddress,
      backgroundColor,
      title,
    } = this.props;

    const [beforeFadeImg, startFadeImg, finishFadeImg] = [
      this.scrollPosition(30),
      this.scrollPosition(40),
      this.scrollPosition(70),
    ];
    const [beforeFadeName, startFadeName, finishFadeName] = [
      this.scrollPosition(50),
      this.scrollPosition(60),
      this.scrollPosition(75),
    ];

    const imageOpacity = this.scrollY.y.interpolate({
      inputRange: [0, beforeFadeImg, startFadeImg, finishFadeImg],
      outputRange: [0, 0, 0.5, 1],
      extrapolate: 'clamp',
    });
    const nameOpacity = this.scrollY.y.interpolate({
      inputRange: [0, beforeFadeName, startFadeName, finishFadeName],
      outputRange: [0, 0, 0.5, 1],
      extrapolate: 'clamp',
    });

    return (
      <View style={[styles.headerWrapper, styles.userModalHeader, { backgroundColor }]}>
        <View style={styles.headerMenu}>
          <TouchableOpacity
            hitSlop={sizes.hitSlop}
            onPress={leftTopIconOnPress}
            style={styles.leftHeaderButton}>
            <Image style={styles.icon} resizeMode="contain" source={leftTopIcon} />
          </TouchableOpacity>
          <View style={styles.headerTitleContainer}>
            <Animated.View style={[styles.headerPic, { opacity: imageOpacity }]}>
              <SvgUri
                width="30"
                height="30"
                uri={`https://avatars.dicebear.com/api/pixel-art/${walletAddress}.svg`}
              />
            </Animated.View>
            <Animated.Text numberOfLines={1} style={[styles.headerTitle, { opacity: nameOpacity }]}>
              {title}
            </Animated.Text>
          </View>
          <TouchableOpacity
            hitSlop={sizes.hitSlop}
            onPress={rightTopIconOnPress}
            style={styles.rightHeaderButton}>
            <Image style={styles.icon} resizeMode="contain" source={rightTopIcon} />
          </TouchableOpacity>
        </View>
      </View>
    );
  };

  renderHeader = () => {
    const { header } = this.props;
    const renderHeader = header || this.renderAvatarHeader;

    return renderHeader();
  };

  renderAvatarForeground = () => {
    const { walletAddress, subtitle, title } = this.props;
    const startSize = constants.responsiveWidth(18);
    const endSize = constants.responsiveWidth(12);

    const [startImgAnimation, finishImgAnimation] = [
      this.scrollPosition(startImgPosition),
      this.scrollPosition(finishImgPosition),
    ];
    const [startAuthorFade, finishAuthorFade] = [this.scrollPosition(40), this.scrollPosition(50)];

    const [startAboutFade, fininshAboutFade] = [this.scrollPosition(60), this.scrollPosition(70)];

    const imageOpacity = this.scrollY.y.interpolate({
      inputRange: [0, startImgAnimation, finishImgAnimation],
      outputRange: [1, 0.8, 0],
      extrapolate: 'clamp',
    });
    const imageSize = this.scrollY.y.interpolate({
      inputRange: [0, startImgAnimation, finishImgAnimation],
      outputRange: [startSize, startSize, endSize],
      extrapolate: 'clamp',
    });
    const authorOpacity = this.scrollY.y.interpolate({
      inputRange: [0, startAuthorFade, finishAuthorFade],
      outputRange: [1, 1, 0],
      extrapolate: 'clamp',
    });
    const aboutOpacity = this.scrollY.y.interpolate({
      inputRange: [0, startAboutFade, fininshAboutFade],
      outputRange: [1, 1, 0],
      extrapolate: 'clamp',
    });

    return (
      <View style={styles.foreground}>
        <Animated.View style={{ opacity: imageOpacity }}>
          <Animated.View style={[styles.profilePic, { width: imageSize, height: imageSize }]}>
            <SvgUri
              width="90"
              height="90"
              uri={`https://avatars.dicebear.com/api/pixel-art/${walletAddress}.svg`}
            />
          </Animated.View>
        </Animated.View>

        <Animated.View
          style={[
            styles.messageContainer,
            styles.userModalMessageContainer,
            { opacity: authorOpacity },
          ]}>
          <Text numberOfLines={2} style={[styles.message, styles.foregroundTitle]}>
            {title}
          </Text>
        </Animated.View>
        <Animated.View style={[styles.infoContainer, { opacity: aboutOpacity }]}>
          <Text style={styles.infoText}>{subtitle}</Text>
        </Animated.View>
      </View>
    );
  };

  renderForeground = () => {
    const { foreground } = this.props;
    const renderForeground = foreground || this.renderAvatarForeground;

    return renderForeground();
  };

  renderBackground = () => {
    const {
      headerLayout: { height },
    } = this.state;
    const { backgroundColor, hasBorderRadius } = this.props;

    const headerBorderRadius = this.scrollY.y.interpolate({
      inputRange: [0, height],
      outputRange: [80, 0],
      extrapolate: 'extend',
    });

    const borderBottomRightRadius = hasBorderRadius ? headerBorderRadius : 0;

    return (
      <Animated.View
        style={[
          styles.background,
          {
            borderBottomRightRadius,
            backgroundColor,
          },
        ]}
      />
    );
  };

  render() {
    const {
      backgroundColor,
      backgroundImage,
      contentContainerStyles,
      children,
      renderBody,
      headerHeight,
      snapToEdge,
      bounces,
      scrollEvent,
      parallaxHeight,
      snapStartThreshold,
      snapStopThreshold,
      snapValue,
      transparentHeader,
      scrollRef,
      keyboardShouldPersistTaps,
      refreshControl,
      onMomentumScrollEnd,
      onMomentumScrollBegin,
    } = this.props;

    if (renderBody) {
      console.warn('Warning: renderBody prop is deprecated. Please use children instead');
    }

    return (
      <>
        <StatusBar backgroundColor={backgroundColor} barStyle="light-content" />
        <StickyParallaxHeader
          contentContainerStyles={contentContainerStyles}
          foreground={this.renderForeground()}
          header={this.renderHeader()}
          deviceWidth={constants.deviceWidth}
          scrollEvent={event([{ nativeEvent: { contentOffset: { y: this.scrollY.y } } }], {
            useNativeDriver: false,
            listener: (e) => scrollEvent && scrollEvent(e),
          })}
          headerSize={this.setHeaderSize}
          headerHeight={headerHeight}
          background={this.renderBackground()}
          backgroundImage={backgroundImage}
          bounces={bounces}
          snapToEdge={snapToEdge}
          parallaxHeight={parallaxHeight}
          transparentHeader={transparentHeader}
          snapStartThreshold={snapStartThreshold}
          snapStopThreshold={snapStopThreshold}
          snapValue={snapValue}
          scrollRef={scrollRef}
          keyboardShouldPersistTaps={keyboardShouldPersistTaps}
          refreshControl={refreshControl}
          onMomentumScrollEnd={onMomentumScrollEnd}
          onMomentumScrollBegin={onMomentumScrollBegin}>
          {renderBody ? renderBody() : children}
        </StickyParallaxHeader>
      </>
    );
  }
}

AvatarHeader.propTypes = {
  hasBorderRadius: bool,
  bounces: bool,
  snapToEdge: bool,
  leftTopIconOnPress: func,
  rightTopIconOnPress: func,
  leftTopIcon: Image.propTypes.source,
  rightTopIcon: Image.propTypes.source,
  backgroundColor: string,
  headerHeight: number,
  backgroundImage: Image.propTypes.source,
  contentContainerStyles: ViewPropTypes.style,
  title: string,
  subtitle: string,
  walletAddress: string,
  children: node,
  renderBody: func,
  scrollEvent: func,
  parallaxHeight: number,
  foreground: func,
  header: func,
  snapStartThreshold: oneOfType([bool, number]),
  snapStopThreshold: oneOfType([bool, number]),
  snapValue: oneOfType([bool, number]),
  transparentHeader: bool,
  scrollRef: oneOfType([func, shape({ current: instanceOf(ScrollView) })]),
  keyboardShouldPersistTaps: oneOf(['never', 'always', 'handled', false, true, undefined]),
  refreshControl: element,
  headerSize: func,
  onMomentumScrollEnd: func,
  onMomentumScrollBegin: func,
};
AvatarHeader.defaultProps = {
  leftTopIconOnPress: () => {},
  rightTopIconOnPress: () => {},
  leftTopIcon: require('../../assets/icons/iconCloseWhite.png'),
  rightTopIcon: require('../../assets/icons/Icon-Menu.png'),
  backgroundColor: Brandon.color,
  headerHeight: sizes.userModalHeaderHeight,
  backgroundImage: null,
  contentContainerStyles: {},
  title: Brandon.author,
  subtitle: Brandon.about,
  walletAddress: 'string',
  children: <RenderContent user={Brandon} />,
  bounces: true,
  snapToEdge: true,
  hasBorderRadius: true,
  parallaxHeight: sizes.userScreenParallaxHeader,
  transparentHeader: false,
  scrollRef: null,
  keyboardShouldPersistTaps: undefined,
  refreshControl: undefined,
  headerSize: undefined,
  onMomentumScrollEnd: undefined,
  onMomentumScrollBegin: undefined,
};

export default AvatarHeader;
