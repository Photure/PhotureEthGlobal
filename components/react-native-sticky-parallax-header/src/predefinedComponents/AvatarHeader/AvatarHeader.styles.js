import { StyleSheet, Platform } from 'react-native';
import { ifIphoneX } from '../../utils';
import { colors, constants, screenStyles } from '../../constants';

export default StyleSheet.create({
  ...screenStyles,
  userModalHeader: {
    paddingTop: Platform.OS === 'ios' ? ifIphoneX(50, 30) : 18,
  },
  headerPic: {
    width: 32,
    height: 32,
    borderRadius: 8,
  },
  headerMenu: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  headerTitleContainer: {
    flex: 8,
    flexDirection: 'row',
    alignItems: 'center',
    marginRight: 24,
  },
  headerTitle: {
    fontSize: 16,
    lineHeight: 20,
    color: colors.white,
    marginLeft: 12,
    marginRight: 12,
  },
  infoContainer: {
    flexDirection: 'row',
    marginBottom: 16,
  },
  iconContainer: {
    flexDirection: 'row',
    padding: 8,
    borderRadius: 8,
    backgroundColor: colors.white,
    width: 56,
  },
  infoText: {
    flexGrow: 1,
    color: colors.white,
    fontSize: constants.normalizedFontSize(16),
    lineHeight: 24,
  },
  icon: {
    width: 16,
    height: 16,
  },
  footerContainer: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    marginHorizontal: 24,
  },
  authorPhoto: {
    width: 32,
    height: 32,
    borderRadius: 8,
  },
  userModalMessageContainer: {
    marginTop: 12,
    paddingBottom: 8,
  },
  rightHeaderButton: {
    flex: 1,
    alignItems: 'flex-end',
  },
  leftHeaderButton: {
    flex: 1,
  },
  foregroundTitle: {
    flexGrow: 1,
    paddingRight: 12,
  },
});
