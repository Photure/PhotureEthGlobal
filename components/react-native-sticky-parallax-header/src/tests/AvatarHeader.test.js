import React from 'react';
import renderer from 'react-test-renderer';
import AvatarHeader from '../predefinedComponents/AvatarHeader/AvatarHeader';

describe('AvatarHeader Test [Hacktoberfest]', () => {
  test('-Snapshot AvatarHeader', () => {
    const tree = renderer.create(<AvatarHeader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
