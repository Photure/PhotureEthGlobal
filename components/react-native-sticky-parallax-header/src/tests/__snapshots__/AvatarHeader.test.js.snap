// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarHeader Test [Hacktoberfest] -Snapshot AvatarHeader 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgb(78,15,255)",
        "height": 75,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 24,
            "paddingTop": 40,
            "width": "100%",
          },
          Object {
            "paddingTop": 30,
          },
          Object {
            "backgroundColor": "rgb(78,15,255)",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 15,
              "left": 15,
              "right": 15,
              "top": 15,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
            }
          }
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../src/assets/icons/iconCloseWhite.png",
              }
            }
            style={
              Object {
                "height": 16,
                "width": 16,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 8,
              "flexDirection": "row",
              "marginRight": 24,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/photosPortraitBrandon.png",
              }
            }
            style={
              Object {
                "borderRadius": 8,
                "height": 32,
                "opacity": 0,
                "width": 32,
              }
            }
          />
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "lineHeight": 20,
                "marginLeft": 12,
                "marginRight": 12,
                "opacity": 0,
              }
            }
          >
            Brandon
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 15,
              "left": 15,
              "right": 15,
              "top": 15,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 1,
              "opacity": 1,
            }
          }
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../src/assets/icons/Icon-Menu.png",
              }
            }
            style={
              Object {
                "height": 16,
                "width": 16,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    bounces={true}
    bouncesZoom={true}
    contentContainerStyle={
      Object {
        "backgroundColor": undefined,
        "minHeight": 1966.02,
      }
    }
    decelerationRate="fast"
    forwardedRef={[Function]}
    nestedScrollEnabled={true}
    onScroll={[Function]}
    onScrollEndDrag={[Function]}
    overScrollMode="never"
    scrollEventThrottle={1}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "height": 707.02,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 300,
                "left": 0,
                "overflow": "scroll",
                "position": "absolute",
                "right": 0,
                "top": -300,
              },
              Object {
                "backgroundColor": "rgb(78,15,255)",
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "justifyContent": "flex-end",
                "left": 0,
                "marginBottom": 5,
                "paddingBottom": 3,
                "position": "absolute",
                "top": 0,
                "width": 750.5,
              },
              Object {
                "height": 707.02,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(78,15,255)",
                "borderBottomRightRadius": 80,
                "height": "100%",
                "justifyContent": "flex-end",
                "width": "100%",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": undefined,
              "height": 707.02,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "flex-end",
                "paddingHorizontal": 24,
              }
            }
          >
            <View
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/assets/images/photosPortraitBrandon.png",
                  }
                }
                style={
                  Object {
                    "borderRadius": 33.75,
                    "height": 135,
                    "width": 135,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "marginTop": 12,
                  "opacity": 1,
                  "paddingBottom": 8,
                  "paddingTop": 24,
                }
              }
            >
              <Text
                numberOfLines={2}
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 75,
                      "letterSpacing": -1,
                      "lineHeight": 90,
                    },
                    Object {
                      "flexGrow": 1,
                      "paddingRight": 12,
                    },
                  ]
                }
              >
                Brandon
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "marginBottom": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "flexGrow": 1,
                    "fontSize": 17,
                    "lineHeight": 24,
                  }
                }
              >
                Coffee buff. Web enthusiast. Unapologetic student. Gamer. Avid organizer.
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RCTScrollView
          alwaysBounceVertical={false}
          automaticallyAdjustContentInsets={false}
          contentContainerStyle={
            Object {
              "minHeight": 551.98,
            }
          }
          contentOffset={
            Object {
              "x": 0,
            }
          }
          directionalLockEnabled={true}
          forwardedRef={[Function]}
          horizontal={true}
          keyboardDismissMode="on-drag"
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          pagingEnabled={true}
          scrollEnabled={false}
          scrollEventThrottle={16}
          scrollsToTop={false}
          showsHorizontalScrollIndicator={false}
        >
          <View>
            <View
              style={
                Array [
                  Object {
                    "maxHeight": null,
                    "minHeight": 551.98,
                    "width": 750,
                  },
                  Object {},
                ]
              }
            >
              <View
                onLayout={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "justifyContent": "center",
                      "paddingBottom": 24,
                      "paddingHorizontal": 24,
                    },
                    Object {
                      "marginBottom": 0,
                      "paddingBottom": 707.02,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "flex-start",
                      "color": "black",
                      "fontSize": 24,
                      "letterSpacing": -0.2,
                      "lineHeight": 28,
                      "paddingBottom": 20,
                      "paddingTop": 40,
                    }
                  }
                >
                  Author's Quizes
                </Text>
                <View
                  accessible={true}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderColor": "rgb(246,245,248)",
                      "borderRadius": 24,
                      "borderWidth": 0,
                      "marginTop": 12,
                      "opacity": 1,
                      "paddingHorizontal": 20,
                      "paddingVertical": 16,
                      "shadowColor": "rgb(35,35,35)",
                      "shadowOffset": Object {
                        "heght": 2,
                        "width": 2,
                      },
                      "shadowOpacity": 0.08,
                      "shadowRadius": 40,
                      "width": "100%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "flex-start",
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "rgb(246,245,248)",
                          "borderRadius": 16,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "rgb(71,71,71)",
                            "fontSize": 12,
                            "lineHeight": 16,
                            "paddingHorizontal": 8,
                            "paddingVertical": 4,
                          }
                        }
                      >
                        Product Design
                      </Text>
                    </View>
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "rgb(78, 15, 255)",
                            "borderRadius": 8,
                            "flexDirection": "row",
                            "padding": 8,
                            "shadowColor": "rgb(35,35,35)",
                            "shadowOffset": Object {
                              "heght": 2,
                              "width": 0,
                            },
                            "shadowOpacity": 0.016,
                            "shadowRadius": 32,
                            "width": 56,
                          },
                          false,
                        ]
                      }
                    >
                      <Image
                        source={
                          Object {
                            "testUri": "../../../src/assets/icons/cards.png",
                          }
                        }
                        style={
                          Object {
                            "height": 16,
                            "width": 16,
                          }
                        }
                      />
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 16,
                              "lineHeight": 20,
                              "paddingLeft": 5,
                            },
                            false,
                          ]
                        }
                      >
                        10
                      </Text>
                    </View>
                  </View>
                  <View
                    style={Object {}}
                  >
                    <Text
                      style={
                        Object {
                          "color": "black",
                          "fontSize": 24,
                          "letterSpacing": -0.2,
                          "lineHeight": 28,
                          "paddingTop": 8,
                        }
                      }
                    >
                      Design System
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      accessible={true}
                      focusable={false}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                          "paddingRight": 40,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "flex-start",
                            "paddingTop": 22,
                          }
                        }
                      >
                        <Image
                          resizeMode="contain"
                          source={
                            Object {
                              "testUri": "../../../src/assets/images/photosPortraitBrandon.png",
                            }
                          }
                          style={
                            Object {
                              "borderRadius": 8,
                              "height": 24,
                              "width": 24,
                            }
                          }
                        />
                        <Text
                          style={
                            Object {
                              "color": "black",
                              "fontSize": 12,
                              "lineHeight": 16,
                              "paddingLeft": 8,
                            }
                          }
                        >
                          Brandon
                        </Text>
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "width": "38%",
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
